cumulus_etl/loaders/i2b2/transform.py

```diff
@@ -15,6 +18,8 @@
 ###############################################################################
 
 
+ICD_CODE_DICT = json.load(open(Path(Path(__file__).resolve().parent,'icd.json')))
```

Comment:
It seemed silly to me to load in a 10MB json dict a million times - again, this could be done someplace else if you prefer.

Reply:
I'm a little sensitive to memory usage - we do a lot of stuff to not keep data in memory if we don't have to. Can we load this in a way that would allow it to be GC'd when done with it?

Obviously don't load it a million times. But load it once per i2b2 load, then null it out. The current code structure doesn't have an obvious place to do that. Do you have thoughts there?

You could kind of hack something for that by adding like init and cleanup methods to `transform.py` that maybe `_load_all_with_extractors` could call? Doesn't feel elegant, but might be easiest.

Another option (which feels more elegant and just as easy?) is to do something like this in `I2b2Loader`:
```
from functools import partial
...
        if "Condition" in resources:
            code_dict = load_big_json_file()
            self._loop(
                conditions(),
                partial(transform.to_fhir_condition, code_dict),
                os.path.join(tmpdir.name, "Condition.ndjson"),
            )
``` 
And change to_fhir_condition to accept a code dict as the first param.

At the end of the day, 10MB ain't the worst. But still.

Reply:
yeah i was originally thinking about kwarg passing, this was just conceptually faster for something with some time pressure - i could go up the call chain and add the dict loader in at that entry point and be explicit about it.

currying functions hurts my brain. if you prefer that i can do it, but that's not normally something i would even think to try in a vacuum.

Reply:
Hah, I thought you loved functional programming and would be all about it. :smile: 

I don't particularly care about `partial`, but I thought it might avoid a larger refunction of that whole `_loop` setup that is otherwise very generic. (though I guess you could just pass `**kwarg` down through, so :shrug: easy either way, as I type this.)

If this is urgent, you can land this and clean up later - 10MB will not break us. But the kwarg approach seems good to me as a cleanup (now or later). Will approve to get out of your way.

Reply:
i think that functional programming is very good for memory usage and once you get past map/reduce i get very fuzzy on low level details, especially since it's been a year or two since i've had to do any, and my brain is a leaky bucket.